File: apps/mobile/src/types/attendance.ts
export interface AttendanceSession {
  id: string;
  event_id: string;
  campus_id: string;
  session_date: string;
  total_count: number;
  male_count: number;
  female_count: number;
  children_count: number;
  new_converts: number;
  first_timers: number;
  check_in_open: boolean;
  opened_at: string;
  closed_at: string | null;
  notes: string | null;
}

export interface CheckInRequest {
  attendance_type: "member" | "guest" | "first_timer";
  member_id?: string;
  guest_id?: string;
  anonymous_name?: string;
  check_in_method?: string;
}

export interface CheckInResponse {
  id: string;
  session_id: string;
  attendance_type: string;
  member_id: string | null;
  guest_id: string | null;
  anonymous_name: string | null;
  check_in_at: string;
  check_in_method: string;
}

File: backend/app/schemas/attendance.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field


class AttendanceSessionCreate(BaseModel):
    event_id: UUID
    session_date: date
    notes: str | None = None


class AttendanceSessionUpdate(BaseModel):
    male_count: int | None = Field(None, ge=0)
    female_count: int | None = Field(None, ge=0)
    children_count: int | None = Field(None, ge=0)
    new_converts: int | None = Field(None, ge=0)
    first_timers: int | None = Field(None, ge=0)
    notes: str | None = None
    check_in_open: bool | None = None


class AttendanceSessionResponse(BaseModel):
    id: UUID
    event_id: UUID
    campus_id: UUID
    session_date: date
    total_count: int
    male_count: int
    female_count: int
    children_count: int
    new_converts: int
    first_timers: int
    check_in_open: bool
    opened_at: datetime
    closed_at: datetime | None
    notes: str | None

    model_config = {"from_attributes": True}


class CheckInRequest(BaseModel):
    attendance_type: str = Field(pattern="^(member|guest|first_timer)$")
    member_id: UUID | None = None
    guest_id: UUID | None = None
    anonymous_name: str | None = None
    check_in_method: str = "manual"


class CheckInResponse(BaseModel):
    id: UUID
    session_id: UUID
    attendance_type: str
    member_id: UUID | None
    guest_id: UUID | None
    anonymous_name: str | None
    check_in_at: datetime
    check_in_method: str

    model_config = {"from_attributes": True}
